scene/main/scene_tree.cpp

```diff
@@ -1336,13 +1354,27 @@ void SceneTree::_call_input_pause(const StringName &p_group, CallInputType p_cal
 		}
 	}
 
-	for (const ObjectID &id : no_context_node_ids) {
-		if (p_viewport->is_input_handled()) {
-			break;
+	if (p_call_type == CALL_INPUT_TYPE_SHORTCUT_INPUT && !p_viewport->is_input_handled()) {
+		// These controls did not have a context, therefore may receive shortcut input from anywhere.
+		// Sort them by their priority in handling the shortcut.
+		// Note: We can only make a best-effort attempt to select the node to handle the input first. To customize
+		// shortcut handling and propagation logic more explicitly, the shortcut context should be used.
+		Vector<Pair<Node *, bool>> resolved_nodes_without_context;
+		for (const Pair<ObjectID, bool> &pair : nodes_without_context) {
+			Node *node = Object::cast_to<Node>(ObjectDB::get_instance(pair.first));
```

Comment:
```suggestion
			Node *node = ObjectDB::get_instance<Node>(pair.first);
```
(you'll need to rebase to use this)